app/controllers/course/discussion/posts_controller.rb

```diff
@@ -7,6 +7,11 @@ class Course::Discussion::PostsController < Course::ComponentController
   include Course::Discussion::PostsConcern
 
   def create
+    # Set parent as the topologically last pre-existing post, if it exists.
+    # @post is in @topic.posts, so we filter out @post, which has no id yet.
+    if @topic.posts.length > 1
+      @post.parent = @topic.posts.ordered_topologically.flatten.select(&:id).last
```

Comment:
`@topic.posts.ordered_topologically.flatten.select(&:id)` will this one always return nil ?  cause the new post is the last ?